# Practice with pandas

Load a dataset of your choice from  https://github.com/fivethirtyeight/data which we neither used in class nor in the group exercise by using 

In [ ]:
import pandas as pd
import numpy as np

df = pd.read_csv('your_url')

(you might have to choose the raw format).
Use the summary command to describe your data. Use at least five further built-in commands from numpy or pandas to obtain information about your data and give a short summary of your findings. 

In [33]:
df = pd.read_csv('https://raw.githubusercontent.com/fivethirtyeight/data/master/bad-drivers/bad-drivers.csv')

In [34]:
df.head()

State  \
0     Alabama   
1      Alaska   
2     Arizona   
3    Arkansas   
4  California   

   Number of drivers involved in fatal collisions per billion miles  \
0                                               18.8                  
1                                               18.1                  
2                                               18.6                  
3                                               22.4                  
4                                               12.0                  

   Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding  \
0                                                 39                      
1                                                 41                      
2                                                 35                      
3                                                 18                      
4                                                 35                      

   Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired  \
0                                                 30                              
1                                                 25                              
2                                                 28                              
3                                                 26                              
4                                                 28                              

   Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted  \
0                                                 96                            
1                                                 90                            
2                                                 84                            
3                                                 94                            
4                                                 91                            

   Percentage Of Drivers Involved In Fatal Collisions Who Had Not Been Involved In Any Previous Accidents  \
0                                                 80                                                        
1                                                 94                                                        
2                                                 96                                                        
3                                                 95                                                        
4                                                 89                                                        

   Car Insurance Premiums ($)  \
0                      784.55   
1                     1053.48   
2                      899.47   
3                      827.34   
4                      878.41   

   Losses incurred by insurance companies for collisions per insured driver ($)  
0                                             145.08                             
1                                             133.93                             
2                                             110.35                             
3                                             142.39                             
4                                             165.63

In [35]:
df.describe()

Number of drivers involved in fatal collisions per billion miles  \
count                                          51.000000                  
mean                                           15.790196                  
std                                             4.122002                  
min                                             5.900000                  
25%                                            12.750000                  
50%                                            15.600000                  
75%                                            18.500000                  
max                                            23.900000                  

       Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding  \
count                                          51.000000                      
mean                                           31.725490                      
std                                             9.633438                      
min                                            13.000000                      
25%                                            23.000000                      
50%                                            34.000000                      
75%                                            38.000000                      
max                                            54.000000                      

       Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired  \
count                                          51.000000                              
mean                                           30.686275                              
std                                             5.132213                              
min                                            16.000000                              
25%                                            28.000000                              
50%                                            30.000000                              
75%                                            33.000000                              
max                                            44.000000                              

       Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted  \
count                                          51.000000                            
mean                                           85.921569                            
std                                            15.158949                            
min                                            10.000000                            
25%                                            83.000000                            
50%                                            88.000000                            
75%                                            95.000000                            
max                                           100.000000                            

       Percentage Of Drivers Involved In Fatal Collisions Who Had Not Been Involved In Any Previous Accidents  \
count                                           51.00000                                                        
mean                                            88.72549                                                        
std                                              6.96011                                                        
min                                             76.00000                                                        
25%                                             83.50000                                                        
50%                                             88.00000                                                        
75%                                             95.00000                                                        
max                                            100.00000                                                        

       Car Insurance Premiums ($)  \
count         

In [36]:
df.shape

(51, 8)

In [37]:
df.columns

Index([u'State',
       u'Number of drivers involved in fatal collisions per billion miles',
       u'Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding',
       u'Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired',
       u'Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted',
       u'Percentage Of Drivers Involved In Fatal Collisions Who Had Not Been Involved In Any Previous Accidents',
       u'Car Insurance Premiums ($)',
       u'Losses incurred by insurance companies for collisions per insured driver ($)'],
      dtype='object')

In [53]:
df.State[df['Car Insurance Premiums ($)']>1250]

8     District of Columbia
18               Louisiana
30              New Jersey
Name: State, dtype: object

In [48]:
df.State[df['Number of drivers involved in fatal collisions per billion miles']==df['Number of drivers involved in fatal collisions per billion miles'].max()]

34      North Dakota
40    South Carolina
Name: State, dtype: object

In [39]:
df.State[df['Number of drivers involved in fatal collisions per billion miles']>40]

Series([], Name: State, dtype: object)

In [40]:
df.median()

Number of drivers involved in fatal collisions per billion miles                                           15.60
Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding                                       34.00
Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired                               30.00
Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted                                 88.00
Percentage Of Drivers Involved In Fatal Collisions Who Had Not Been Involved In Any Previous Accidents     88.00
Car Insurance Premiums ($)                                                                                858.97
Losses incurred by insurance companies for collisions per insured driver ($)                              136.05
dtype: float64

In [30]:
df['Number of drivers involved in fatal collisions per billion miles'].sum()

805.2999999999998

In [41]:
df.corr()

Number of drivers involved in fatal collisions per billion miles  \
Number of drivers involved in fatal collisions ...                                           1.000000                  
Percentage Of Drivers Involved In Fatal Collisi...                                          -0.029080                  
Percentage Of Drivers Involved In Fatal Collisi...                                           0.199426                  
Percentage Of Drivers Involved In Fatal Collisi...                                           0.009782                  
Percentage Of Drivers Involved In Fatal Collisi...                                          -0.017942                  
Car Insurance Premiums ($)                                                                  -0.199702                  
Losses incurred by insurance companies for coll...                                          -0.036011                  

                                                    Percentage Of Drivers Involved In Fatal Collisions Who Were Speeding  \
Number of drivers involved in fatal collisions ...                                          -0.029080                      
Percentage Of Drivers Involved In Fatal Collisi...                                           1.000000                      
Percentage Of Drivers Involved In Fatal Collisi...                                           0.286244                      
Percentage Of Drivers Involved In Fatal Collisi...                                           0.131738                      
Percentage Of Drivers Involved In Fatal Collisi...                                           0.014066                      
Car Insurance Premiums ($)                                                                   0.042541                      
Losses incurred by insurance companies for coll...                                          -0.061241                      

                                                    Percentage Of Drivers Involved In Fatal Collisions Who Were Alcohol-Impaired  \
Number of drivers involved in fatal collisions ...                                           0.199426                              
Percentage Of Drivers Involved In Fatal Collisi...                                           0.286244                              
Percentage Of Drivers Involved In Fatal Collisi...                                           1.000000                              
Percentage Of Drivers Involved In Fatal Collisi...                                           0.043380                              
Percentage Of Drivers Involved In Fatal Collisi...                                          -0.245455                              
Car Insurance Premiums ($)                                                                  -0.017451                              
Losses incurred by insurance companies for coll...                                          -0.083916                              

                                                    Percentage Of Drivers Involved In Fatal Collisions Who Were Not Distracted  \
Number of drivers involved in fatal collisions ...                                           0.009782                            
Percentage Of Drivers Involved In Fatal Collisi...                                           0.131738                            
Percentage Of Drivers Involved In Fatal Collisi...                                           0.043380                            
Percentage Of Drivers Involved In Fatal Collisi...                                           1.000000                            
Percentage Of Drivers Involved In Fatal Collisi...                                          -0.195265                            
Car Insurance Premiums ($)                                                                   0.019578                            
Losses incurred by insurance companies for coll...                                          -0.058467                            



### Summary
The dataset describes the number of fatal car collisions per driven distance in each US state, possible causes for the accidents and related insurance costs. Most fatal collisions happen in North Dakota and South Carolina, the total number of fatal accidents in the US is 805 per billion miles. There is a high correlation between car insurance premium and losses incurred by insurance companies for collision per insured driver.